pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -208,14 +211,19 @@ func (rc *reconciler) reconcile(ctx context.Context) {
 				continue
 			}
 			// Check whether timeout has reached the maximum waiting time
-			timeout := elapsedTime > rc.maxWaitForUnmountDuration
+			drainTimeoutExpired := elapsedTime > rc.maxWaitForUnmountDuration
 
 			isHealthy, err := rc.nodeIsHealthy(attachedVolume.NodeName)
 			if err != nil {
 				logger.Error(err, "Failed to get health of node", "node", klog.KRef("", string(attachedVolume.NodeName)))
 			}
 
-			// Force detach volumes from unhealthy nodes after maxWaitForUnmountDuration.
+			// Force detach volumes from unhealthy nodes after maxWaitForUnmountDuration if force detach is enabled
+			// Ensure that the timeout condition checks this correctly so that the correct metric is updated below
+			timeout := drainTimeoutExpired && !rc.disableForceDetachOnTimeout
```

Comment:
```suggestion
			forceDetatchTimeoutExpired := drainTimeoutExpired && !rc.disableForceDetachOnTimeout
```

To clarify what this timeout is throughout this function (it's rather long)?